### PR TITLE
Item checkout test tff 128

### PR DESCRIPTION
**Pull request by Cafferoq**
### Summary

Functional tests for checking out an item

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Pull
2. Install and run the tests.
3. Verify the tests work and make sense

### Links

- [TFF-128](https://msoese.atlassian.net/browse/TFF-128)

### CODE REVIEW

File: test/functional/item-checkout.js [hunk 1]
```diff
@@ -92,51 +86,313 @@ describe('item checkout', function () {
...
+    it('checks out the item', () => {
+
+        mockServer.expect({
+            method: 'get',
+            endpoint: '/api/item',
+            request: {
+                address: 'iGwEZUvfA'
+            },
+            response:{
+                "status":"success",
```
Comment (mattmahn) [inline test/functional/item-checkout.js]: What is this, JSON?

File: test/functional/item-checkout.js [hunk 2]
```diff
@@ -92,51 +86,313 @@ describe('item checkout', function () {
...
+                    ],
+                    email: 'vonneumann@msoe.edu',
+                    major: 'Chemical Engineering & Mathematics'
                 }
             }
         });
-        return app.client.keys('iGwEZUvfA').then(() => {
+
+
+        return app.client.waitForVisible('.cart input[type="text"]').then(() => {
```
Comment (mattmahn) [inline test/functional/item-checkout.js]: How did the test get to the cart page without entering a student ID?

File: test/functional/item-checkout.js (hunk 1 above)
Comment (Cafferoq) [inline test/functional/item-checkout.js]: The response that the mock server sends when it receives the expected call.

Comment (Cafferoq) [inline test/functional/item-checkout.js]: The mock server's response when the expected call is made to the server.

Comment (mattmahn) [inline test/functional/item-checkout.js]: I understand, it's moreso "Why is there JSON in the Javascript, get rid of those quotes".

**Pull request merged**